### PR TITLE
Arc - register certain CDI components under BuildContext so extension…

### LINKED ISSUE #2188: ArC - add BuildContext.Key.SCOPES so that extensions can work with the final set of scope types
We could also add `Key.QUALIFIERS`, `Key.INTERCEPTOR_BINDINGS` and possibly others.

### PR DESCRIPTION
**Pull request by manovotn**
…s can use that.
Modify SpringDIProcessor to make use of this.

Resolves #2188

### CODE REVIEW

**Comment by geoand** [comment]
:+1: for the Spring parts

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java [hunk 1]
```diff
@@ -146,6 +146,12 @@
             buildContext.putInternal(Key.INJECTION_POINTS.asString(), Collections.unmodifiableList(injectionPoints));
             buildContext.putInternal(Key.OBSERVERS.asString(), Collections.unmodifiableList(observers));
             buildContext.putInternal(Key.BEANS.asString(), Collections.unmodifiableList(beans));
+            List<ScopeInfo> allScopes = BuiltinScope.getAllBuiltInScopes();
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java]: It's rather minor but I don't like this kind of contract. I think that `BuiltinScope.getAllBuiltInScopes()` should either return an immutable list... or even better, just create a new list and then `Collections.addAll(scopes, BuiltinScope.values())`, no need for extra method.

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java]: Ah, you actually need `ScopeInfo`. In that case, I would try with a stream ;-).

Comment (manovotn) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java]: In fact I dont really 'need' `ScopeInfo`, the only use case we have is in the Spring bits (that I know of) and there you need `DotName`. But I found it more useful to provide ScopeInfo as that seems more resourceful than just name.
I can do it the stream way.

**Comment by manovotn** [comment]
Also, looks like I broke some other Spring bits with it, will check the failing tests in the morning.

**Comment by geoand** [comment]
Let me know if you need any help with the Spring tests

**Comment by manovotn** [comment]
I think I found it. Looks like `AnnotationsTransformers` could be invoked before I populated the build context with `SCOPES`. Should be easy enough to swap that around...

**Review by mkouba — APPROVED** [review]

**Review by geoand — APPROVED** [review]

**Pull request merged**